hoomd/operations.py

```diff
@@ -117,29 +213,76 @@ def integrator(self, op):
 
     @property
     def updaters(self):
+        """list[`hoomd.operation.Updater`]: A list of updater operations.
+
+        Holds the list of updaters associated with this collection. The list can
+        be modified as a standard Python list.
+        """
         return self._updaters
 
     @property
     def analyzers(self):
+        """list[`hoomd.operation.Analzyer`]: A list of analyzer operations.
+
+        Holds the list of analyzers associated with this collection. The list
+        can be modified as a standard Python list.
+        """
         return self._analyzers
 
     @property
     def tuners(self):
+        """list[`hoomd.operation.Tuner`]: A list of tuner operations.
+
+        Holds the list of tuners associated with this collection. The list can be
+        modified as a standard Python list.
+        """
         return self._tuners
 
     def __iadd__(self, operation):
+        """Works the same as `Operations.add`.
+
+        Args:
+            operation (``operation``): A HOOMD-blue updater, analyzers, compute,
+                tuner, or integrator to add to the object.
+        """
         self.add(operation)
+        return self
 
     def remove(self, operation):
+        """Remove an operation from the `Operations` object.
+
+        Remove the item from the collection whose id is the same as
+        ``operation``. See
+        `<https://docs.python.org/3/library/functions.html#id>`_ for the concept
+        of a Python object id.
+
+        Args:
+            operation (``operation``): A HOOMD-blue integrator, updater,
+                analyzer, tuner, or compute, to remove from the container.
+
+        Raises:
+            ValueError: raises if operation is not found in this container.
+            TypeError: raises if operation is not of a valid type for this
+                container.
+        """
         if isinstance(operation, hoomd.integrate._BaseIntegrator):
-            raise ValueError(
-                "Cannot remove iterator without setting to a new integator.")
-        elif isinstance(operation, _Analyzer):
+            self.integrator = None
+        elif isinstance(operation, Analyzer):
```

Comment:
I haven't looked to see how many times this long `elif` pattern is repeated, but it might be worth having some kind of dispatch mechanism:
```python
# Integrators are probably treated differently...
if isinstance(operation, hoomd.integrate._BaseIntegrator):
    # stuff
# ...but all the other operations have a lot of copy-paste that could be eliminated
elif isinstance(operation, _Operation):
    container = self._container_for_operation_type(operation)
    container.remove(operation)

...

def _container_for_operation_type(self, operation):
    if isinstance(operation, Analyzer):
        return self._analyzers
    elif ...
```